lib/rust/lexer/definition/src/lexer.rs

```diff
@@ -464,9 +722,805 @@ impl EnsoLexer {
+        let offset     = self.offset.consume();
+        let bad_quotes = self.consume_current();
+        let token      = Token::InvalidQuote(bad_quotes,offset);
+        self.append_token(token);
+    }
+
+    /// Submit a missing quote in a nested text line literal.
+    fn submit_missing_quote_nested<R:LazyReader>
+    ( &mut self
+      , line_ending:token::LineEnding
+      , reader:&mut R
+    ) {
+        trace!(self.logger,"Text::submit_missing_quote_nested");
+        let interpolate_is_closed = false;
+        let matched_bookmark      = self.bookmarks.matched_bookmark;
+        self.bookmarks.rewind(matched_bookmark,reader);
+        let nested         = self.text_state.unsafe_current_mut();
+        nested.style       = Some(token::TextStyle::UnclosedLine);
+        let nested_text    = self.text_state.unsafe_end_literal();
+        let nested_literal = nested_text.into();
+        self.append_token(nested_literal);
+        self.pop_state();
+        self.on_interpolate_end(interpolate_is_closed,reader);
+        let top_level       = self.text_state.unsafe_current_mut();
+        let top_level_style = top_level.style.expect("Must be set.");
+        top_level.style = match top_level_style {
+            token::TextStyle::FormatLine => Some(token::TextStyle::UnclosedLine),
+            _                            => Some(top_level_style),
+        };
+        let text    = self.text_state.unsafe_end_literal();
+        let literal = text.into();
+        self.append_token(literal);
+        self.pop_state();
+        match line_ending {
+            token::LineEnding::None => self.on_eof(reader),
+            _                       => self.block_state.push_line_ending(line_ending),
+        }
+    }
+
+    /// Submit a missing closing quote in a text line literal.
+    fn submit_missing_quote<R:LazyReader>(&mut self, new_line:token::LineEnding, reader:&mut R) {
+        trace!(self.logger,"Text::submit_missing_quote");
+        let matched_bookmark = self.bookmarks.matched_bookmark;
+        self.bookmarks.rewind(matched_bookmark, reader);
+        let current   = self.text_state.unsafe_current_mut();
+        current.style = Some(token::TextStyle::UnclosedLine);
+        let text      = self.text_state.unsafe_end_literal();
+        let literal   = text.into();
+        self.append_token(literal);
+        self.pop_state();
+        match new_line {
+            token::LineEnding::None => self.on_eof(reader),
+            _                       => self.block_state.push_line_ending(new_line),
+        }
+    }
+
+    /// Triggered when encountering the opening of a text block in a nested context.
+    fn on_text_block_nested<R:LazyReader>(&mut self, new_line:token::LineEnding, reader:&mut R) {
+        trace!(self.logger,"Text::on_text_block_nested");
+        let current      = self.consume_current();
+        let offset       = self.offset.consume();
+        let text         = &current[0..3];
```

Comment:
these literals are magic. What if we change the number of openning quotes? This should be corelated / exported somewhere as a constant